### PR TITLE
Onboard temperature

### PR DESCRIPTION
**Pull request by fkuhne**
Adjust code according to comments on previous PR.

### CODE REVIEW

**Comment by lurch** [comment]
Looks like you need to rebase and force-push.

File: adc/onboard_temperature/onboard_temperature.c [hunk 1]
```diff
@@ -1,18 +1,26 @@
+/**
+ * Copyright (c) 2020 Raspberry Pi (Trading) Ltd.
```
Comment (lurch) [inline adc/onboard_temperature/onboard_temperature.c:2]: 2021 :wink:

**Comment by fkuhne** [comment]
Rebase done.

**Comment by lurch** [comment]
The "Commits" tab still shows 11 commits, which doesn't look right? :confused: 

This PR LGTM otherwise :+1:

**Comment by fkuhne** [comment]
Oh, I see. Well, I tried to do this PR by myself but clearly I'm stumbling. Truth is that although being a git user for year, I've never done a PR before :).

@lurch, can you point me out the best way to do it? Should I close this one and start from the beginning?

Thanks!

**Comment by lurch** [comment]
It _should_ just be something like:
```
git fetch origin
git rebase origin/develop
git push -f fork onboard_temperature
```

See https://docs.github.com/en/github/using-git/using-advanced-git-commands for more help. Or if you get in a complete mess you could just close this PR and create a fresh one :laughing:

**Comment by fkuhne** [comment]
Sounds better now, with the exception of `usb/device/dev_hid_generic_inout/dev_hid_generic_inout.c` which appeared here with some typo fixes.

**Comment by lurch** [comment]
Looks like #30 got merged straight into the `master` branch but somehow never made it onto the `develop` branch :confused: (which is probably why it's "hanging around" on your branch now that you've rebased your PR from `master` to `develop`)

So you can either:
 * get fancy with interactive-rebase to remove that commit from your branch
 * kill your current branch and create a new one
 * wait for #77 to get merged, hopefully after which rebasing onto develop will work correctly

Sorry for the confusion! :slightly_frowning_face:

**Comment by fkuhne** [comment]
OK, I think that's it.

I'm thinking in, maybe, contribute with this repo with some more examples. I would go for Tensorflow Lite Micro integration but it will take time. There is something you guys see that I can do to help?

**Comment by lurch** [comment]
> OK, I think that's it.

Almost... the "Commits" tab now lists 6 commits, with the last commit being the "inverse" of the first commit. Would you mind squashing them down to a single commit please? (I think that would make the "bad commits" disappear)
Again, sorry this has been so awkward for you!

> I would go for Tensorflow Lite Micro integration but it will take time.

That's already been done at https://github.com/raspberrypi/pico-tflmicro :slightly_smiling_face: 

> There is something you guys see that I can do to help?

We're a bit busy at the moment getting the next SDK-version out of the door, but once that's done I'll ask the team if they have any ideas. Thanks for the offer of help :+1:

**Comment by fkuhne** [comment]
> > OK, I think that's it.
> 
> Almost... the "Commits" tab now lists 6 commits, with the last commit being the "inverse" of the first commit. Would you mind squashing them down to a single commit please? (I think that would make the "bad commits" disappear)
> Again, sorry this has been so awkward for you!
> 

I have to say sorry to bother all this time for such a silly thing... It should not be that harsh, haha. But that's OK, it was an opportunity to learn.

Maybe it's better to start a fresh new PR. If I squash the commits the username will be from the guy who fixed #30. It's not because I want my name, but I think it may cause confusion later... What do you think? Either way I'm OK with your decision.

> > I would go for Tensorflow Lite Micro integration but it will take time.
> 
> That's already been done at https://github.com/raspberrypi/pico-tflmicro 🙂
> 

That's awesome! I'l go there right away get my hands dirty. :D

> > There is something you guys see that I can do to help?
> 
> We're a bit busy at the moment getting the next SDK-version out of the door, but once that's done I'll ask the team if they have any ideas. Thanks for the offer of help 👍

My pleasure! ;)

**Comment by lurch** [comment]
> But that's OK, it was an opportunity to learn.

Yup, always good to be learning :+1: 

> If I squash the commits the username will be from the guy who fixed #30

True, I hadn't considered that!

> Maybe it's better to start a fresh new PR.

Yeah, that might be the easiest option in the end!!

**Comment by lurch** [comment]
@fkuhne #77 has now been merged, so with a bit of luck, and if the wind is blowing in the right direction, one more round of:
```
git fetch origin
git rebase origin/develop
git push -f fork onboard_temperature
```
_should_ get things back in order? :crossed_fingers:

**Pull request closed without merging**